File: Teoria-PO/Aula 2/verificador_ordenacao.py
def esta_ordenada(array):
    """
    Verifica se uma lista está ordenada em ordem crescente.
    
    Args:
        array: lista de elementos a ser verificada
        
    Returns:
        True se estiver ordenada, False caso contrário
    """
    if array is None or len(array) <= 1:
        return True 

    for i in range(len(array) - 1): 
        if array [i] > array[i + 1]:
            return False 
         
        return True 

if __name__ == "__main__": 
    array_ordenado = [1, 2, 3, 4, 5]
    array_desordenado = [5, 2, 1, 4, 3]

    print("Array Ordenado: ", esta_ordenada(array_ordenado))
    print("Array Desordenado: ", esta_ordenada(array_desordenado))
    

File: Teoria-Po/Aula 2/VerificadorOrdenacao.java
public class VerificadorOrdenacao {
    
    /**
     * Verifica se um array de inteiros está ordenado em ordem crescente.
     * @param array o array a ser verificado
     * @return true se estiver ordenado, false caso contrário
     */
    public static boolean estaOrdenado(int[] array) {
        // Um array vazio ou com um único elemento é considerado ordenado
        if (array == null || array.length <= 1) {  // lenght serve para saber o tamanho de um array 
            return true;
        }
        
        for (int i = 0; i < array.length - 1; i++) {
            if (array[i] > array[i + 1]) {
                return false;
            }
        }
        
        return true;
    }
    
    public static void main(String[] args) {
        int[] arrayOrdenado = {1, 2, 3, 4, 5};
        int[] arrayDesordenado = {5, 3, 1, 4, 2};
        
        System.out.println("Array ordenado: " + estaOrdenado(arrayOrdenado));       // true
        System.out.println("Array desordenado: " + estaOrdenado(arrayDesordenado)); // false
    }
}